.github/workflows/scripts/scanner-update-nvd.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python3
+
+import os
+import json
+from datetime import datetime, timedelta
+import logging
+from urllib import request, parse
+import time
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s: %(levelname)s: %(message)s")
+
+BASE_URL = os.getenv('SCANNER_NVD_URL', "https://services.nvd.nist.gov/rest/json/cves/2.0")
+API_KEY = os.getenv('SCANNER_NVD_API_KEY', None)
+
+
+def get_dates(year, interval):
+    # interval should be a factor of 12
+    if 12 % interval != 0:
+        raise ValueError("Interval must be a factor of 12.")
+    current_date = datetime.now()
+
+    for month in range(1, 13, interval):
+        start = datetime(year, month, 1)
+        # return if start date is in the future
+        if start > current_date:
+            return
+
+        # Handling for months greater than 12 (for intervals like 4, 6 months, etc.)
+        end_month = month + interval
+        end_year = year
+        if end_month > 12:
+            end_month -= 12
+            end_year += 1
+
+        end = datetime(end_year, end_month, 1) - timedelta(days=1)
+        yield start.strftime("%Y-%m-%dT00:00:00.000"), end.strftime("%Y-%m-%dT23:59:59.999")
+
+
+def fetch_data(start, end):
+    index, total = 0, 0
+    logging.info(f"Fetching page from {start} to {end}")
+    backoff_time = 1
```

Comment:
I would consider bumping this to 10s, so it would wait up to minutes when exponentially retrying. Do you know if `request.urplopen` raise if HTTP status is different from 200 or it will be the `json.loads()` that will raise because the `response.read().decode()` will not be a valid JSON?

Reply:
I think http status other than 200 and json errors will be caught in that try exception block. Not sure if this is what's concerning you.

Reply:
Yes, I wanted to confirm if `urlopen()` raises or returns a response which you need to check for statuses.